src/DotVVM.Framework/Compilation/DefaultViewCompiler.cs

```diff
@@ -35,7 +35,7 @@ public DefaultViewCompiler(DotvvmConfiguration configuration)
         /// <summary>
         /// Compiles the view and returns a function that can be invoked repeatedly. The function builds full control tree and activates the page.
         /// </summary>
-        public virtual CSharpCompilation CompileView(string sourceCode, string fileName, CSharpCompilation compilation, string namespaceName, string className)
+        public virtual (ControlBuilderDescriptor, Func<CSharpCompilation>) CompileView(string sourceCode, string fileName, CSharpCompilation compilation, string namespaceName, string className)
```

Comment:
I don't have a problem with tuples returned inside non public methods, but this method is public and virtual. 
We should return an object here, because someone may need to override this function and return an object with additional properties (if the method returns object, he can return something that extends it). Using tuple in virtual methods breaks the extensibility totally.